[Bug]: Error: The terraform-provider-aws_v4.34.0_x5 plugin crashed!

### Terraform Core Version

>1.1

### AWS Provider Version

4.34.0

### Affected Resource(s)

I'm trying to apply the aws_ssm_service_setting resource to an account to enable high throughput for ssm parameter store but out pipeline keeps crashing with an error related to the aws plugin 

Error: Plugin did not respond
│ 
│   with aws_ssm_service_setting.ssm_throughput_setting,
│   on resources-ssm.tf line 28, in resource "aws_ssm_service_setting" "ssm_throughput_setting":
│   28: resource "aws_ssm_service_setting" "ssm_throughput_setting" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.

### Expected Behavior

That the pipeline would go through and that the setting would be applied 

### Actual Behavior

Pipeline crrashes 

### Relevant Error/Panic Output Snippet

```shell
Terraform will perform the following actions:
  # aws_ssm_service_setting.ssm_throughput_setting will be created
  + resource "aws_ssm_service_setting" "ssm_throughput_setting" {
      + arn           = (known after apply)
      + id            = (known after apply)
      + setting_id    = "arn:aws:ssm:eu-west-1:*:servicesetting/ssm/parameter-store/high-throughput-enabled"
      + setting_value = "true"
      + status        = (known after apply)
    }
Plan: 1 to add, 0 to change, 0 to destroy.
aws_ssm_service_setting.ssm_throughput_setting: Creating...
╷
│ Error: Plugin did not respond
│ 
│   with aws_ssm_service_setting.ssm_throughput_setting,
│   on resources-ssm.tf line 28, in resource "aws_ssm_service_setting" "ssm_throughput_setting":
│   28: resource "aws_ssm_service_setting" "ssm_throughput_setting" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵
Stack trace from the terraform-provider-aws_v4.34.0_x5 plugin:
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x7643ee9]
goroutine 387 [running]:
github.com/hashicorp/terraform-provider-aws/internal/service/ssm.statusServiceSetting.func1()
	github.com/hashicorp/terraform-provider-aws/internal/service/ssm/status.go:61 +0x69
github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource.(*StateChangeConf).WaitForStateContext.func1()
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.23.0/helper/resource/state.go:110 +0x207
created by github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource.(*StateChangeConf).WaitForStateContext
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.23.0/helper/resource/state.go:83 +0x1d8
Error: The terraform-provider-aws_v4.34.0_x5 plugin crashed!
This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
time=2022-10-13T08:48:17Z level=error msg=Terraform invocation failed in /tmp/.terragrunt-cache/deploy/regional/ssm/prod/<account_id>/eu-west-1/0PcmPd7RdhvjPhjJEtvg5nsCsjU/80Rk9KJRp-bTumAoWpzI88KbbbE prefix=[deploy/regional/ssm] 
time=2022-10-13T08:48:17Z level=error msg=1 error occurred:
	* exit status 1
```


### Terraform Configuration Files

terraform {
  backend "s3" {}
  required_version = "~> 1.1"
  required_providers {
    aws = "~> 4.0"
  }
}

### Steps to Reproduce

Add a resource to apply the setting to the account e.g.

resource "aws_ssm_service_setting" "ssm_throughput_setting" {
  setting_id    = "arn:aws:ssm:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:servicesetting/ssm/parameter-store/high-throughput-enabled"
  setting_value = "true"
}

### Debug Output

_No response_

### Panic Output

_No response_

### Important Factoids

no

### References

_No response_

### Would you like to implement a fix?

_No response_

--- a/.changelog/27232.txt
+++ b/.changelog/27232.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+resource/aws_ssm_service_setting: Prevent panic during status read
+```

--- a/internal/service/ssm/find.go
+++ b/internal/service/ssm/find.go
@@ -87,9 +87,9 @@ func FindPatchGroup(conn *ssm.SSM, patchGroup, baselineId string) (*ssm.PatchGro
 	return result, err
 }
 
-func FindServiceSettingByARN(conn *ssm.SSM, arn string) (*ssm.ServiceSetting, error) {
+func FindServiceSettingByID(conn *ssm.SSM, id string) (*ssm.ServiceSetting, error) {
 	input := &ssm.GetServiceSettingInput{
-		SettingId: aws.String(arn),
+		SettingId: aws.String(id),
 	}
 
 	output, err := conn.GetServiceSetting(input)

--- a/internal/service/ssm/find.go
+++ b/internal/service/ssm/find.go
@@ -99,7 +99,7 @@ func FindServiceSettingByARN(conn *ssm.SSM, arn string) (*ssm.ServiceSetting, er
 	}
 
 	if output == nil || output.ServiceSetting == nil {
-		return nil, fmt.Errorf("finding %s: empty result", arn)
+		return nil, tfresource.NewEmptyResultError(input)
 	}
 
 	return output.ServiceSetting, nil

--- a/internal/service/ssm/find.go
+++ b/internal/service/ssm/find.go
@@ -94,6 +94,13 @@ func FindServiceSettingByID(conn *ssm.SSM, id string) (*ssm.ServiceSetting, erro
 
 	output, err := conn.GetServiceSetting(input)
 
+	if tfawserr.ErrCodeContains(err, ssm.ErrCodeServiceSettingNotFound) {
+		return nil, &resource.NotFoundError{
+			LastError:   err,
+			LastRequest: input,
+		}
+	}
+
 	if err != nil {
 		return nil, err
 	}

--- a/internal/service/ssm/service_setting.go
+++ b/internal/service/ssm/service_setting.go
@@ -74,7 +74,7 @@ func resourceServiceSettingRead(d *schema.ResourceData, meta interface{}) error 
 
 	log.Printf("[DEBUG] Reading SSM Activation: %s", d.Id())
 
-	output, err := FindServiceSettingByARN(conn, d.Id())
+	output, err := FindServiceSettingByID(conn, d.Id())
 	if err != nil {
 		return create.Error(names.SSM, create.ErrActionReading, ResNameServiceSetting, d.Id(), err)
 	}

--- a/internal/service/ssm/service_setting_test.go
+++ b/internal/service/ssm/service_setting_test.go
@@ -5,7 +5,6 @@ import (
 	"testing"
 
 	"github.com/aws/aws-sdk-go/aws"
-	"github.com/aws/aws-sdk-go/aws/awserr"
 	"github.com/aws/aws-sdk-go/service/ssm"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
@@ -13,6 +12,7 @@ import (
 	"github.com/hashicorp/terraform-provider-aws/internal/conns"
 	"github.com/hashicorp/terraform-provider-aws/internal/create"
 	tfssm "github.com/hashicorp/terraform-provider-aws/internal/service/ssm"
+	"github.com/hashicorp/terraform-provider-aws/internal/tfresource"
 	"github.com/hashicorp/terraform-provider-aws/names"
 )
 
@@ -57,26 +57,35 @@ func testAccServiceSettingDestroy(s *terraform.State) error {
 			continue
 		}
 
-		output, err := conn.GetServiceSetting(&ssm.GetServiceSettingInput{
-			SettingId: aws.String(rs.Primary.Attributes["setting_id"]),
-		})
-		_, ok := err.(awserr.Error)
-		if !ok {
+		output, err := tfssm.FindServiceSettingByID(conn, rs.Primary.Attributes["setting_id"])
+
+		if tfresource.NotFound(err) {
+			continue
+		}
+
+		if err != nil {
 			return err
 		}
-		if output.ServiceSetting.Status != aws.String("default") {
-			return create.Error(names.SSM, create.ErrActionCheckingDestroyed, tfssm.ResNameServiceSetting, rs.Primary.Attributes["setting_id"], err)
+
+		if aws.StringValue(output.Status) == "Default" {
+			continue
 		}
+
+		return create.Error(names.SSM, create.ErrActionCheckingDestroyed, tfssm.ResNameServiceSetting, rs.Primary.Attributes["setting_id"], err)
 	}
 
 	return nil
 }
 
-func testAccServiceSettingExists(n string, res *ssm.ServiceSetting) resource.TestCheckFunc {
+func testAccServiceSettingExists(n string, v *ssm.ServiceSetting) resource.TestCheckFunc {
 	return func(s *terraform.State) error {
 		rs, ok := s.RootModule().Resources[n]
 		if !ok {
 			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No SSM Service Setting ID is set")
 		}
 
 		conn := acctest.Provider.Meta().(*conns.AWSClient).SSMConn
@@ -87,7 +96,7 @@ func testAccServiceSettingExists(n string, res *ssm.ServiceSetting) resource.Tes
 			return create.Error(names.SSM, create.ErrActionReading, tfssm.ResNameServiceSetting, rs.Primary.Attributes["setting_id"], err)
 		}
 
-		*res = *output
+		*v = *output
 
 		return nil
 	}

--- a/internal/service/ssm/service_setting_test.go
+++ b/internal/service/ssm/service_setting_test.go
@@ -81,7 +81,7 @@ func testAccServiceSettingExists(n string, res *ssm.ServiceSetting) resource.Tes
 
 		conn := acctest.Provider.Meta().(*conns.AWSClient).SSMConn
 
-		output, err := tfssm.FindServiceSettingByARN(conn, rs.Primary.Attributes["setting_id"])
+		output, err := tfssm.FindServiceSettingByID(conn, rs.Primary.Attributes["setting_id"])
 
 		if err != nil {
 			return create.Error(names.SSM, create.ErrActionReading, tfssm.ResNameServiceSetting, rs.Primary.Attributes["setting_id"], err)

--- a/internal/service/ssm/status.go
+++ b/internal/service/ssm/status.go
@@ -58,6 +58,10 @@ func statusServiceSetting(conn *ssm.SSM, arn string) resource.StateRefreshFunc {
 			return nil, "", nil
 		}
 
+		if err != nil {
+			return nil, "", err
+		}
+
 		return output, aws.StringValue(output.Status), nil
 	}
 }

--- a/internal/service/ssm/status.go
+++ b/internal/service/ssm/status.go
@@ -50,9 +50,9 @@ func statusDocument(conn *ssm.SSM, name string) resource.StateRefreshFunc {
 	}
 }
 
-func statusServiceSetting(conn *ssm.SSM, arn string) resource.StateRefreshFunc {
+func statusServiceSetting(conn *ssm.SSM, id string) resource.StateRefreshFunc {
 	return func() (interface{}, string, error) {
-		output, err := FindServiceSettingByARN(conn, arn)
+		output, err := FindServiceSettingByID(conn, id)
 
 		if tfresource.NotFound(err) {
 			return nil, "", nil

--- a/internal/service/ssm/wait.go
+++ b/internal/service/ssm/wait.go
@@ -71,11 +71,11 @@ func waitDocumentActive(conn *ssm.SSM, name string) (*ssm.DocumentDescription, e
 	return nil, err
 }
 
-func waitServiceSettingUpdated(conn *ssm.SSM, arn string, timeout time.Duration) (*ssm.ServiceSetting, error) {
+func waitServiceSettingUpdated(conn *ssm.SSM, id string, timeout time.Duration) (*ssm.ServiceSetting, error) {
 	stateConf := &resource.StateChangeConf{
 		Pending: []string{"PendingUpdate", ""},
 		Target:  []string{"Customized", "Default"},
-		Refresh: statusServiceSetting(conn, arn),
+		Refresh: statusServiceSetting(conn, id),
 		Timeout: timeout,
 	}
 
@@ -88,11 +88,11 @@ func waitServiceSettingUpdated(conn *ssm.SSM, arn string, timeout time.Duration)
 	return nil, err
 }
 
-func waitServiceSettingReset(conn *ssm.SSM, arn string, timeout time.Duration) error {
+func waitServiceSettingReset(conn *ssm.SSM, id string, timeout time.Duration) error {
 	stateConf := &resource.StateChangeConf{
 		Pending: []string{"Customized", "PendingUpdate", ""},
 		Target:  []string{"Default"},
-		Refresh: statusServiceSetting(conn, arn),
+		Refresh: statusServiceSetting(conn, id),
 		Timeout: timeout,
 	}
 

